Index: llama.cpp/CMakeLists.txt
===================================================================
--- llama.cpp.orig/CMakeLists.txt	2024-03-12 17:04:04.689586131 -0500
+++ llama.cpp/CMakeLists.txt	2024-03-12 17:04:23.088997153 -0500
@@ -331,7 +331,7 @@
 
         add_compile_definitions(GGML_USE_OPENBLAS)
 
-        if (${BLAS_INCLUDE_DIRS} MATCHES "mkl" AND (${LLAMA_BLAS_VENDOR} MATCHES "Generic" OR ${LLAMA_BLAS_VENDOR} MATCHES "Intel"))
+        if ((${BLAS_INCLUDE_DIRS} MATCHES "mkl" AND ${LLAMA_BLAS_VENDOR} MATCHES "Generic") OR ${LLAMA_BLAS_VENDOR} MATCHES "Intel")
             add_compile_definitions(GGML_BLAS_USE_MKL)
         endif()
 
